vscode-extension/src/providers/textDocumentContentProvider.ts

```diff
@@ -0,0 +1,49 @@
+/**
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import * as path from 'path';
+import * as fs from 'fs';
+import {TextDocumentContentProvider, Event, Uri, ProviderResult} from 'vscode';
+import {RelayExtensionContext} from '../context';
+
+const EMPTY_JSON_SCHEMA = '{}';
+const RELAY_CONFIG_SCHEMA_PATH = 'relay-config-schema';
+const PACKAGE_JSON_RELAY_CONFIG_SCHEMA_PATH =
+  'package-json-relay-config-schema';
+
+export class RelayTextDocumentContentProvider
+  implements TextDocumentContentProvider
+{
+  private readonly jsonSchemaPath: string;
+
+  constructor(context: RelayExtensionContext) {
+    this.jsonSchemaPath = path.join(
+      context.relayBinaryExecutionOptions.binaryPath,
```

Comment:
If the user changes their `binaryPath` in the config, do we correectly recompute this?

Reply:
No, I should change this. I was thinking that when the path is changed the entire extension must reload anyways, but that's not true, since the path is just used to start the compiler once, if it's changed after that it doesn't affect the system.